pkg/controller/acmechallenges/sync.go

```diff
@@ -256,14 +267,20 @@ func (c *controller) handleFinalizer(ctx context.Context, ch *cmacme.Challenge)
 	if len(ch.Finalizers) == 0 {
 		return nil
 	}
-	if ch.Finalizers[0] != cmacme.ACMEFinalizer {
+	if otherFinalizerPresent(ch) {
```

Comment:
This logic looks odd to me. Why are we just checking the first finalizer now? And I don't think we need the `otherFinalizerPresent` function. Why do we care if the resource has another finalizer here?

Reply:
@munnerz added the relevant code in #1116

https://github.com/cert-manager/cert-manager/pull/1116/commits/dd8f98768b3211b924155ce15ca0088162e16155

Reply:
I don't understand the need for this logic. Kubernetes will only finalize the resource deletion when all finalizers are gone: https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/. We should only have to care for "our" finalizers. If a foreign controller has added a finalizer, that controller should remove its finalizer.

Reply:
The only reason I can think of is that the ordering would be important, but I doubt that is the case.

Reply:
Assuming the current code is wrong, what should it be doing?

Is it trying to see if _any_ other finalizer is in any slot of the list? Or is it trying to see if our finalizer isn't in the list? Or should the logic be removed?

The code change here is the _closest_ to the spirit of the current code, but it sure doesn't seem like the current behavior is desired.

Reply:
@munnerz do you remember why you added this check?

Reply:
I would collapse this conditional with the previous one: Exit early if none of our finalizers are present.

````
if !finalizerRequired(ch) {
  return nil
}
````

Maybe @munnerz remembers why?